rudder-web/src/main/scala/com/normation/rudder/web/model/DirectiveFieldEditors.scala

```diff
@@ -754,29 +592,54 @@ class MasterPasswordField(val id: String, blankable:Boolean, algos:Seq[HashAlgoC
       s.updateValue(keepCurrentPwd, blankPwd, pastValue, newInput, chosenAlgo)
     }
 
-    currentValue = newInput
-
+    currentValue = Some(newInput)
     if(keepCurrentPwd) pastValue
     else if(blankPwd) ""
     else if(newInput == "") ""
     else chosenAlgo.serialize(newInput.getBytes())
   }
 
-
-  def parseClient(s: String): Unit = {
-    if (null == s) _x = "" else _x = s
+  /* This will parse the json received from the form
+   * format is:
+   * { "action"  : delete/keep/change
+   *   "hash"    : hash type used
+   *   "password : new password, or undefined/missing field if deleting
+   * }
+   *
+   */
+  def parseClient(s: String) = {
+    import net.liftweb.json._
+    val json = parse(s)
```

Comment:
Seems we are sending exception in other fields ... will do the same, to complicated to change everything